sqlglot/transforms.py

```diff
@@ -431,6 +431,32 @@ def move_ctes_to_top_level(expression: exp.Expression) -> exp.Expression:
     return expression
 
 
+def ensure_bools(expression: exp.Expression) -> exp.Expression:
+    def _ensure_bool(node: exp.Expression) -> None:
+        if (
+            node.is_number
+            or node.is_type(exp.DataType.Type.UNKNOWN, *exp.DataType.NUMERIC_TYPES)
+            or (isinstance(node, exp.Column) and not node.type)
+        ):
+            node.replace(node.neq(0))
+
+    for node, *_ in expression.walk():
+        if isinstance(node, exp.Connector):
+            _ensure_bool(node.left)
+            _ensure_bool(node.right)
+        elif isinstance(node, exp.Not):
+            _ensure_bool(node.this)
+        # We can't replace num in CASE x WHEN num ..., because it's not the full predicate
+        elif isinstance(node, exp.If) and not (
+            isinstance(node.parent, exp.Case) and node.parent.this
```

Comment:
It's repeated here as well

Reply:
this is not the same as what's in simplify

Reply:
it's the same as the ensure bool predicates one though right? https://github.com/tobymao/sqlglot/pull/2543#discussion_r1388446273